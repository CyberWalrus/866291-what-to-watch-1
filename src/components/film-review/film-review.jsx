import React, {Fragment, PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {getReviews} from "../../store/data/selectors.js";
import {Operation as OperationData} from "../../store/data/data.js";

class FilmReview extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentWillMount() {
    this.props.onLoadReviews(this.props.filmId);
  }
  render() {
    return (
      <Fragment>
        <div className="movie-card__reviews movie-card__row">
          <div className="movie-card__reviews-col">
            {this.props.reviews &&
              this.props.reviews.map(
                  ({comment, rating, userName, date, dateHTML}, index) => (
                    <div className="review" key={index}>
                      <blockquote className="review__quote">
                        <p className="review__text">{comment}</p>

                        <footer className="review__details">
                          <cite className="review__author">{userName}</cite>
                          <time className="review__date" dateTime={dateHTML}>
                            {date}
                          </time>
                        </footer>
                      </blockquote>

                      <div className="review__rating">{rating}</div>
                    </div>
                  )
              )}
          </div>
        </div>
      </Fragment>
    );
  }
}

FilmReview.propTypes = {
  filmId: PropTypes.number.isRequired,
  onLoadReviews: PropTypes.func.isRequired,
  reviews: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        comment: PropTypes.string.isRequired,
        rating: PropTypes.string.isRequired,
        userId: PropTypes.number.isRequired,
        userName: PropTypes.string.isRequired,
        date: PropTypes.string.isRequired,
        dateHTML: PropTypes.string.isRequired
      })
  )
};
const mapStateToProps = (state, ownProps) =>
  Object.assign({}, ownProps, {
    reviews: getReviews(state)
  });

const mapDispatchToProps = (dispatch) => ({
  onLoadReviews: (filmId) => dispatch(OperationData.loadReviews(filmId))
});
export {FilmReview};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(FilmReview);
