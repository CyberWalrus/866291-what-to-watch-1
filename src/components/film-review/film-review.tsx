import * as React from "react";
import {PureComponent, Fragment} from "react";
import {connect} from "react-redux";
import {getReviews} from "../../store/data/selectors";
import {Operation as OperationData} from "../../store/data/data";
import {Review} from "../../type/data";

interface Props {
  filmId: number;
  onLoadReviews: (filmId: number) => void;
  reviews: Review[];
}

class FilmReview extends PureComponent<Props, null> {
  constructor(props) {
    super(props);
  }

  componentWillMount() {
    this.props.onLoadReviews(this.props.filmId);
  }
  render() {
    return (
      <Fragment>
        <div className="movie-card__reviews movie-card__row">
          <div className="movie-card__reviews-col">
            {this.props.reviews &&
              this.props.reviews.map(
                  ({comment, rating, userName, date, dateHTML}, index) => (
                    <div className="review" key={index}>
                      <blockquote className="review__quote">
                        <p className="review__text">{comment}</p>

                        <footer className="review__details">
                          <cite className="review__author">{userName}</cite>
                          <time className="review__date" dateTime={dateHTML}>
                            {date}
                          </time>
                        </footer>
                      </blockquote>

                      <div className="review__rating">{rating}</div>
                    </div>
                  )
              )}
          </div>
        </div>
      </Fragment>
    );
  }
}

const mapStateToProps = (state, ownProps) =>
  Object.assign({}, ownProps, {
    reviews: getReviews(state)
  });

const mapDispatchToProps = (dispatch) => ({
  onLoadReviews: (filmId) => dispatch(OperationData.loadReviews(filmId))
});
export {FilmReview};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(FilmReview);
