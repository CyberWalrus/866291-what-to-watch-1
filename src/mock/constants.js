const FilmRoute = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};
const GENRE_DEFOULT = `All genres`;
const NUMBER_FILM = 20;

export {FilmRoute, GENRE_DEFOULT, NUMBER_FILM};

