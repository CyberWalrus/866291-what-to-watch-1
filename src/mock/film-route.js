const FilmRoute = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};
export default FilmRoute;
