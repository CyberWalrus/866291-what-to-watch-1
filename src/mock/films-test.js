const films = [
  {
    id: 1,
    genre: `drama`,
    title: `Aviator`,
    src: `img/aviator.jpg`
  },
  {
    id: 2,
    genre: `drama`,
    title: `Macbeth`,
    src: `img/macbeth.jpg`
  }
];
export default films;
