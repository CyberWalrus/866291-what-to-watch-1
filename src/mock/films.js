const films = [
  {
    id: 1,
    genre: `drama`,
    title: `Aviator`,
    src: `img/aviator.jpg`
  },
  {
    id: 2,
    genre: `drama`,
    title: `Macbeth`,
    src: `img/macbeth.jpg`
  },
  {
    id: 3,
    genre: `drama`,
    title: `Revenant`,
    src: `img/revenant.jpg`
  },
  {
    id: 4,
    genre: `comedy`,
    title: `Johnny English`,
    src: `img/johnny-english.jpg`
  },
  {
    id: 5,
    genre: `comedy`,
    title: `What We Do in the Shadows`,
    src: `img/what-we-do-in-the-shadows.jpg`
  }
];
export default films;
