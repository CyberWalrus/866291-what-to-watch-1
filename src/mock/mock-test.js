const FILMS = [
  {
    bgColor: `#977461`,
    description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer, who escaped from a hospital for the criminally insane.`,
    director: `Martin Scorsese`,
    genre: `Thriller`,
    id: 1,
    isFavorite: false,
    pageUrl: `Shutter Island`,
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    srcPreviewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: `8,1`,
    ratingLevel: `Very good`,
    released: 2010,
    runTime: 138,
    scoresCount: 1002557,
    srcBgImage: `https://es31-server.appspot.com/wtw/static/film/background/Shutter_Island.jpg`,
    srcPosterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Shutter_Island.jpg`,
    srcPreviewImage: `https://es31-server.appspot.com/wtw/static/film/preview/shutter-island.jpg`,
    starrings: [(`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`)],
    title: `Shutter Island`
  },
  {
    bgColor: `#F1E9CE`,
    description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
    director: `Justin Kurzel`,
    genre: `Drama`,
    id: 2,
    isFavorite: false,
    pageUrl: `Macbeth`,
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    srcPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: `6,6`,
    ratingLevel: `Good`,
    released: 2015,
    runTime: 113,
    scoresCount: 48798,
    srcBgImage: `https://es31-server.appspot.com/wtw/static/film/background/Macbeth.jpg`,
    srcPosterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Macbeth.jpg`,
    srcPreviewImage: `https://es31-server.appspot.com/wtw/static/film/preview/macbeth.jpg`,
    starrings: [(`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`)],
    title: `Macbeth`
  }
];
const FILM = {
  bgColor: `#977461`,
  description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer, who escaped from a hospital for the criminally insane.`,
  director: `Martin Scorsese`,
  genre: `Thriller`,
  id: 1,
  isFavorite: false,
  pageUrl: `Shutter Island`,
  srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  srcPreviewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: `8,1`,
  ratingLevel: `Very good`,
  released: 2010,
  runTime: 138,
  scoresCount: 1002557,
  srcBgImage: `https://es31-server.appspot.com/wtw/static/film/background/Shutter_Island.jpg`,
  srcPosterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Shutter_Island.jpg`,
  srcPreviewImage: `https://es31-server.appspot.com/wtw/static/film/preview/shutter-island.jpg`,
  starrings: [(`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`)],
  title: `Shutter Island`
};
const REVIEWS = [
  {
    comment: `test test test test test test test test test test test test test`,
    date: `2019-05-11`,
    dateHTML: `May 11, 2019`,
    id: 1,
    rating: `8,1`,
    userId: 1,
    userName: `Walrus`
  },
  {
    comment: `test test test test test test test test test test test test test`,
    date: `2019-05-11`,
    dateHTML: `May 11, 2019`,
    id: 1,
    rating: `8,1`,
    userId: 1,
    userName: `Walrus`
  }
];

const GENRES = [
  `All genres`,
  `Thriller`,
  `Drama`,
  `Adventure`,
  `Comedy`,
  `Crime`,
  `Action`,
  `Fantasy`
];
export {FILMS, FILM, GENRES, REVIEWS};
