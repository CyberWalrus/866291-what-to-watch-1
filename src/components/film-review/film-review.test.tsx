import * as React from "react";
import * as Enzyme from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import {shallow} from "enzyme";
import {FilmReview} from "./film-review";
import {FILM, REVIEWS} from "../../mock/data-mock";

Enzyme.configure({adapter: new Adapter()});
it(`FilmReview correctly renders after relaunch`, () => {
  const handleClick = jest.fn();
  const tree = toJson(
    shallow(
      <FilmReview
        filmId={FILM.id}
        onLoadReviews={handleClick}
        reviews={REVIEWS}
      />
    )
  );
  expect(tree).toMatchSnapshot();
});
