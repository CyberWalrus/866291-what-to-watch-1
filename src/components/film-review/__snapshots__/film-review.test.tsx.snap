// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmReview correctly renders after relaunch 1`] = `
<Fragment>
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
        key="0"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            test test test test test test test test test test test test test
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Walrus
            </cite>
            <time
              className="review__date"
              dateTime="May 11, 2019"
            >
              2019-05-11
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8,1
        </div>
      </div>
    </div>
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
        key="1"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            test test test test test test test test test test test test test
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Walrus
            </cite>
            <time
              className="review__date"
              dateTime="May 11, 2019"
            >
              2019-05-11
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8,1
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
